duckduckgo/macos-browser
# Logins+ filtering and sorting

Task/Issue URL: https://app.asana.com/0/1199230911884351/1201524927333647/f
Tech Design URL:
CC: @brindy 

**Description**:

This PR adds filtering and sorting menus to the Logins+ feature. It supports filtering by each data types, and sorting by Title, Date Created, and Date Modified. It also supports switching between ascending and descending sort order.

**Steps to test this PR**:
1. Run a build of this branch on an existing install with Logins+ data and make sure it looks good
1. Try filtering by each data type
1. Check that the default empty state still appears when you have no data
1. Check that the empty state updates correctly when adding and deleting items
1. Check that section headers are correct, particularly for dates
1. Check that filtering via the search field works and shows data as you would expect

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
